arch/x86/deflate_quick.c

```diff
@@ -92,11 +92,11 @@ static inline long compare258(const unsigned char *const src0, const unsigned ch
         "cmp        $256 + 16, %[ax]\n\t"
         "jb         1b\n\t"
 
-#ifdef X86
+# if !defined(__x86_64)
```

Comment:
Without this change I see a fail when configuring for 32-bit x86:
```
$ ./configure --32; make -j
[...]
arch/x86/deflate_quick.c: Assembler messages:
arch/x86/deflate_quick.c:90: Error: `movzwq' is only supported in 64-bit mode
```

Reply:
For me it looks there is underlying problem in the configure script as it should override ARCH when "--32" is specified on command line.

Reply:
The error seems to be in the way we detect GCC_ARCH in configure:
```
GCC_ARCH=`$CC $CFLAGS -dumpmachine | sed 's/-.*//g'`
```
Here is the output of gcc and clang:
```
$ gcc  -m32 -std=c99 -dumpmachine 
x86_64-linux-gnu
$ clang  -m32 -std=c99 -dumpmachine 
i386-pc-linux-gnu
```
It looks like it is a known problem: https://gcc.gnu.org/ml/gcc-help/2018-05/msg00053.html
I think a good solution is to check whether the compiler is gcc in which case we should do:
```
$ gcc -print-multiarch
x86_64-linux-gnu
$ gcc -print-multiarch -m32
i386-linux-gnu
```
I will prepare a separate patch that fixes this problem.

Reply:
@sebpop The patch should also handle cases where the compiler doesn't support ```-print-multiarch```.